guardrails/applications/text2sql.py

```diff
@@ -122,7 +127,12 @@ def _init_guard(
         schema_file: Optional[str] = None,
         rail_spec: Optional[str] = None,
         rail_params: Optional[Dict] = None,
-        reask_prompt: str = REASK_PROMPT,
+        reask_messages: list[Dict[str, str]] = [
+            {
+                "role": "user",
+                "content": REASK_PROMPT,
+            }
+        ],
```

Comment:
Pre-initializing lists in _init functions can have odd behaviors sometimes.

```
def foo(my_list=[]):
  my_list.append(len(my_list))
  return my_list

>>> foo()
[0]
>>> foo()
[0, 1]
>>> foo()
[0, 1, 2]
```

If we're modifying and appending to reask_messages internally, we should instead default this to None and check inside the method if it's None.